src/main/scala/edu/ie3/simona/agent/participant/evcs/EvcsAgentFundamentals.scala

```diff
@@ -357,42 +353,188 @@ protected trait EvcsAgentFundamentals
+
+    val evcsModel = getEvcsModel(modelBaseStateData)
+
+    val (tickInterval, lastEvs) =
+      getTickIntervalAndLastEvs(tick, modelBaseStateData)
+
+    validateArrivals(lastEvs, arrivingEvs, evcsModel.chargingPoints)
+
+    val relevantData =
+      EvcsRelevantData(
+        tickInterval,
+        lastEvs
+      )
+
+    val updatedStateData =
+      if (tickInterval > 0) {
+        // if we haven't had any departing EVs for this tick,
+        // this also means that we have not caught up with
+        // calculating the current SOC
+
+        val voltage =
+          getAndCheckNodalVoltage(modelBaseStateData, tick)
+
+        // calculate power with evs that have been parked up until now
+        val (result, evModelsCalculated) =
+          evcsModel.calculatePowerAndEvSoc(
+            tick,
+            voltage,
+            relevantData
+          )
+
+        val updatedRelevantData = relevantData.copy(
+          currentEvs = evModelsCalculated ++ arrivingEvs
+        )
+
+        updateValueStoresInformListeners(
+          modelBaseStateData,
+          tick,
+          result,
+          updatedRelevantData
+        )
+      } else {
+        // if some EVs were departing at the current tick,
+        // we're already up-to-date in that regard
+
+        val updatedRelevantData = relevantData.copy(
+          currentEvs = lastEvs ++ arrivingEvs
+        )
+
+        val updatedRelevantDataStore =
+          ValueStore.updateValueStore(
+            modelBaseStateData.calcRelevantDateStore,
+            tick,
+            updatedRelevantData
+          )
+
+        modelBaseStateData.copy(
+          calcRelevantDateStore = updatedRelevantDataStore
+        )
+      }
+
+    goToIdleReplyCompletionAndScheduleTriggerForNextAction(
+      updatedStateData,
+      scheduler
+    )
+  }
+
+  /** Update the result and calc relevant data value stores and inform all
+    * registered listeners
+    *
+    * @param baseStateData
+    *   The base state data of the collection state
+    * @param tick
+    *   The current tick
+    * @param result
+    *   Result of simulation
+    * @param relevantData
+    *   Data, that have been relevant to this calculation
+    * @return
+    *   Desired state change
+    */
+  private final def updateValueStoresInformListeners(
+      baseStateData: ParticipantModelBaseStateData[
+        ApparentPower,
+        EvcsRelevantData,
+        EvcsModel
+      ],
+      tick: Long,
+      result: ApparentPower,
+      relevantData: EvcsRelevantData
+  ): ParticipantModelBaseStateData[
+    ApparentPower,
+    EvcsRelevantData,
+    EvcsModel
+  ] = {
+    /* Update the value stores */
+    val updatedValueStore =
+      ValueStore.updateValueStore(
+        baseStateData.resultValueStore,
+        tick,
+        result
+      )
+    val updatedRelevantDataStore =
+      ValueStore.updateValueStore(
+        baseStateData.calcRelevantDateStore,
```

Comment:
the `calcRelevantDateStore` should be a "data" and not a "date" store

Reply:
I agree, but 1. I wouldn't change it in this PR and 2. the `calcRelevantDateStore` will be gone with the introduction of flex and stateful agents